includes/class-wp-document-revisions-admin.php

```diff
@@ -884,6 +890,83 @@ public function profile_update_cb() {
 		}
 	}
 
+	/**
+	 * Ensures that any system limit on revisions does not apply to documents
+	 *
+	 * @since 3.2.2
+	 *
+	 * @param int     $num  default value for the number of revisions for the post_type
+	 * @param WP_Post $post current post
+	 */
+	public function manage_document_revisions( $num, $post ) {
+
+		if ( ! is_object( $post ) || 'document' !== $post->post_type ) {
+			return $num;
+		}
+
+		// Set default number as unlimited
+		/**
+		 * Filters the number of revisions to keep for documents.
+		 *
+		 * This should normally be unlimited and setting it can make attachments unaccessible.
+		 *
+		 * @since 3.2.2
+		 *
+		 * @param int -1 (unlimited)
+		 */
+		$num = apply_filters( 'document_max_revisions', -1 );
+
+		return $num;
+
+	}
+
+
+	/**
+	 * Ensures that an error box appears if the revisions for a post has reached a system limit
+	 *
+	 * @since 3.2.2
+	 */
+	public function check_document_revisions() {
+
+		global $post;
+
+		if ( ! is_object( $post ) || 'document' !== $post->post_type ) {
+			return;
+		}
+
+		$num = wp_revisions_to_keep( $post );
+
+		if ( 0 === $num ) {
+			// setting revisions to 0 makes no sense for this plugin
+			// @codingStandardsIgnoreLine WordPress.XSS.EscapeOutput.OutputNotEscaped
+			echo '<div class="notice notice-error"><p>';
+			esc_html_e( 'Maximum number of revisions set to zero using a local filter. Check your configuration.', 'wp-document-revisions' );
+			// @codingStandardsIgnoreLine WordPress.XSS.EscapeOutput.OutputNotEscaped
+			echo '</p></div>';
+
+		} elseif ( 0 < $num ) {
+			// need to check that we're not at the limit
+			$revisions = count( wp_get_post_revisions( $post->ID ) );
+
+			if ( $num < $revisions ) {
```

Comment:
Are the conditionals and messages switched here? I'd think `num == revision` would be maximum revisions reached, and and `num < revisions` to be more revisions exist than permitted?

Reply:
Thank you for your review. I'll make the changes once this is sorted.

Once implemented (with a user defined limit), and only when the user makes a change will any existing version be deleted. 

So once the user has `num` versions, then any change will leave the user with `num` versions.

But if there already exists more than `num` versions , it only displays `num` versions but the others are still there - until an update is made when they will be removed.

So this can occur when the limit is changed (or first implemented).

At one level, an update will delete existing data, so they could be combined into a single `$num <= $version`  warning